jitsi-media-transform/src/main/kotlin/org/jitsi/nlj/rtp/codec/av1/Av1DDPacket.kt

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.nlj.rtp.codec.av1
+
+import org.jitsi.nlj.RtpEncodingDesc
+import org.jitsi.nlj.RtpLayerDesc
+import org.jitsi.nlj.rtp.ParsedVideoPacket
+import org.jitsi.rtp.rtp.RtpPacket
+import org.jitsi.rtp.rtp.header_extensions.Av1DependencyDescriptorHeaderExtension
+import org.jitsi.rtp.rtp.header_extensions.Av1DependencyDescriptorReader
+import org.jitsi.rtp.rtp.header_extensions.Av1DependencyDescriptorStatelessSubset
+import org.jitsi.rtp.rtp.header_extensions.Av1DependencyException
+import org.jitsi.rtp.rtp.header_extensions.Av1TemplateDependencyStructure
+import org.jitsi.rtp.rtp.header_extensions.FrameInfo
+import org.jitsi.utils.logging2.Logger
+
+/** A video packet carrying an AV1 Dependency Descriptor.  Note that this may or may not be an actual AV1 packet;
+ * other video codecs can also carry the AV1 DD.
+ */
+class Av1DDPacket : ParsedVideoPacket {
+    var descriptor: Av1DependencyDescriptorHeaderExtension?
+    val statelessDescriptor: Av1DependencyDescriptorStatelessSubset
+    val frameInfo: FrameInfo?
+    val av1DDHeaderExtensionId: Int
+
+    private constructor(
+        buffer: ByteArray,
+        offset: Int,
+        length: Int,
+        av1DDHeaderExtensionId: Int,
+        encodingId: Int,
+        descriptor: Av1DependencyDescriptorHeaderExtension?,
+        statelessDescriptor: Av1DependencyDescriptorStatelessSubset,
+        frameInfo: FrameInfo?
+    ) : super(buffer, offset, length, encodingId) {
+        this.descriptor = descriptor
+        this.statelessDescriptor = statelessDescriptor
+        this.frameInfo = frameInfo
+        this.av1DDHeaderExtensionId = av1DDHeaderExtensionId
+    }
+
+    constructor(
```

Comment:
VideoQualityLayerLookup (which sets encodingId) is after VideoParser (which calls this constructor) in the receiver pipeline, so that shouldn't happen.  Do you think it's worth preserving this anyway, for robustness?

Reply:
Probably not